src/travel-details-screens/components/Trip.tsx

```diff
@@ -25,19 +29,37 @@ export type TripProps = {
 const Trip: React.FC<TripProps> = ({
   tripPattern,
   error,
+  onPressDetailsMap,
   onPressDeparture,
   onPressQuay,
 }) => {
   const styles = useStyle();
   const legs = tripPattern.legs.filter((leg, i) =>
     isSignificantFootLegWalkOrWaitTime(leg, tripPattern.legs[i + 1]),
   );
+
+  const realtimeMapEnabled = useRealtimeMapEnabled();
+  const shouldShowLive =
+    !tripPattern.legs.find((a) => !a.realtime) && realtimeMapEnabled;
+
+  const ids = shouldShowLive
+    ? (tripPattern.legs
+        .map((leg) => leg.serviceJourney?.id)
+        .filter(Boolean) as string[])
```

Comment:
Casting should be avoided if possible. If the type of `serviceJourney.id` changes (although unlikely) then this code may crash at runtime since it won't give any compile time error when casting to string.

You can take a look at the options here:
https://www.typescriptlang.org/docs/handbook/advanced-types.html

A more practical example is in this blog post:
https://dev.to/zirkelc/how-to-filter-values-type-safely-29jj

Reply:
Great! Will look at this